packages/playground/src/weblets/profile_manager.vue

```diff
@@ -284,13 +287,93 @@ defineProps({
 });
 defineEmits<{ (event: "update:modelValue", value: boolean): void }>();
 
+let mountedTimeout: any;
+watch(
+  () => props.modelValue,
+  m => {
+    if (m) {
+      if (mountedTimeout) {
+        clearTimeout(mountedTimeout);
+      }
+      mountedTimeout = setTimeout(() => {
+        mounted();
+      });
+    }
+  },
+);
+
+function mounted() {
+  if (isStoredCredentials()) {
+    activeTab.value = 0;
+    const credentials: Credentials = getCredentials();
+    const sessionPassword = sessionStorage.getItem("password");
+
+    if (!sessionPassword) return;
+
+    password.value = sessionPassword;
+
+    if (credentials.passwordHash) {
+      return login();
+    }
+  } else {
+    activeTab.value = 1;
+    return;
+  }
+}
+
+function onKeyDown() {
+  if (password.value.length >= 6) {
+    if (activeTab.value === 0) [login()];
+    else {
+      storeAndLogin();
+    }
+  }
+}
+
+function getCredentials() {
+  const getCredentials = localStorage.getItem("wallet");
+  let credentials: Credentials = {};
+
+  if (getCredentials) {
+    credentials = JSON.parse(getCredentials);
+  }
+  return credentials;
+}
```

Comment:
```suggestion

const version = 1; // for example
const WALLET_KEY = "wallet.v" + version
function getCredentials() {
  const getCredentials = localStorage.getItem(WALLET_KEY);
  let credentials: Credentials = {};

  if (getCredentials) {
    credentials = JSON.parse(getCredentials);
  }
  return credentials;
}
```

Support versioning for wallet as we might change how we manage password and mnemonic from time to time.
let's imagine if we changed the storing way when the user open his browser it will be broken as we will be expecting another way of parsing data and here versioning comes into play it won't respect any wallet beside the new one

Reply:
let's assume that we changed the version of storing the wallet, in that case, we have to update the `version`, which is the way we're going to update the `wallet` key, in both cases we have to change the key, what is the difference?

Reply:
Updating the version should be done whenever we change the behaviour of the wallet so it will break the user browser
@AhmedHanafy725

Reply:
the version will make it easier to migrate from the old key to the new one. if the same key is used, you will have to check which one is used first and then decide whether to migrate to the new one or not.